### PR TITLE
Fix for NullReference Bug during GetAzureRoleAssignmentsCommand

### PR DESCRIPTION
**Pull request by danybeam**
## Description

This PR will check if a RoleDefinitionName is null before comparing it.
This is done during the GetAzureRoleAssignmentsCommand

## Checklist

- [X] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [X] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Resources/Resources.Test/ScenarioTests/RoleAssignmentTests.cs [hunk 1]
```diff
@@ -159,6 +159,13 @@ public void RaGetByScope()
             TestRunner.RunTestScript("Test-RaGetByScope");
         }
 
+        [Fact(Skip = "Needs recording. PS-VSPrompt link and TestFramework are broken at time of commit")]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void RaGetOnlyByRoleDefinitionName()
+        {
+            TestRunner.RunTestScript("Test-RaGetOnlyByRoleDefinitionName");
```
Comment (markcowl) [inline src/Resources/Resources.Test/ScenarioTests/RoleAssignmentTests.cs]: Please do not skip a running test.  Please fix any tests that need to be re-recorded, this is one mechanism we use to protect customers from breaking changes.

File: src/Resources/Resources.Test/ScenarioTests/RoleAssignmentTests.ps1 [hunk 2]
```diff
@@ -639,6 +639,21 @@ function Test-RaGetByScope
     VerifyRoleAssignmentDeleted $newAssignment1
 }
 
+<#
```
Comment (darshanhs90) [inline src/Resources/Resources.Test/ScenarioTests/RoleAssignmentTests.ps1]: what is the use of this test?

Comment (danybeam) [reply src/Resources/Resources.Test/ScenarioTests/RoleAssignmentTests.ps1]: Test-RaGetOnlyByRoleDefinitionName test whether the cmdlet is able to retrieve role assignments only with the Role definition name without any additional options

File: src/Resources/Resources.Test/ScenarioTests/RoleAssignmentTests.cs (hunk 1 above)
Comment (danybeam) [reply src/Resources/Resources.Test/ScenarioTests/RoleAssignmentTests.cs]: Un-skipped
recorded new test
re-recorded old tests

**Comment by danybeam** [comment]
@markcowl bump

**Pull request closed without merging**